### PR TITLE
Load summary file from correct location for MicroXS.from_model

### LINKED ISSUE #2172: Can't link `summary.h5` file when using `Model.init_lib()` for running `MicroXS.from_model()`
I am trying to run a validation simulation for the new transport-independent depletion feature on HPC architecture. To get all the right MPI intracommunicator, I am running `Model.init_lib()` before passing my `Model` object to `MicroXS.from_model()`. This produces the following error:

```bash
Traceback (most recent call last):
  File "/home/ooblack/projects/2023-openmc-depletion-data/problem_setup.py", line 75, in <module>
    micro_xs = MicroXS.from_model(model, materials[0], chain_file)
  File "/home/ooblack/projects/openmc/openmc/deplete/microxs.py", line 108, in from_model
    xs[rx].load_from_statepoint(sp)
  File "/home/ooblack/projects/openmc/openmc/mgxs/mgxs.py", line 994, in load_from_statepoint
    raise ValueError(msg)
ValueError: Unable to load data from a statepoint which has not been linked with a summary file
```
I believe that this error is caused by our use of a `TemporaryDirectory` in `from_model()` to avoid overwriting the user's `material.xml` and `tallies.xml` files. On initialization, the `StatePoint` object tries to link to a `summary.h5` file that is assumed to be in the same directory as the statepoint file. When using just the python API, both the statepoint file and the summary file are written to the current working directory. Howver, after calling `init_lib()`, the statepoint file is written to 
the temporary directory, but the summary file is still written to the original current working directory.

The `settings.output.path` variable provides a fix provided that the variable is set before calling `Model.init_lib()`. Unfortunately, this use-case has the user calling `Model.init_lib()` before the `from_model()` function.

This issue can be closed when a PR exists that adds machinery to `MicroXS.from_model()` that allows error-free use of a model that has the shared library initialized.

### PR DESCRIPTION
**Pull request by paulromano**
Closes #2172. @yardasol can you see if this works for your use case?

### CODE REVIEW

**Comment by yardasol** [comment]
It runs, but now I get the following error:
```bash
Traceback (most recent call last):
  File "/home/ooblack/projects/2023-openmc-depletion-data/problem_setup.py", line 75, in <module>
    micro_xs = MicroXS.from_model(model, materials[0], chain_file)
  File "/home/ooblack/projects/openmc/openmc/deplete/microxs.py", line 109, in from_model
    xs[rx].load_from_statepoint(sp)
  File "/home/ooblack/projects/openmc/openmc/mgxs/mgxs.py", line 1036, in load_from_statepoint
    sp_tally = statepoint.get_tally(
  File "/home/ooblack/projects/openmc/openmc/statepoint.py", line 639, in get_tally
    raise LookupError('Unable to get Tally')
LookupError: Unable to get Tally
```

I have had success with re-initializing the shared library using the following:
```python
            if model.is_initialized:
                model.settings.output = {'path': temp_dir,
                                         'summary': True,
                                         'tallies': False}
                model.init_lib()
```
but I'd prefer not

File: openmc/deplete/microxs.py [hunk 1]
```diff
@@ -97,7 +96,15 @@ def from_model(cls,
             run_kwargs.setdefault('cwd', temp_dir)
             statepoint_path = model.run(**run_kwargs)
 
-            with StatePoint(statepoint_path) as sp:
+            # Load summary object -- if model.init_lib() was called, the
+            # summary.h5 file is in the current working directory
+            if model.is_initialized:
+                summary = Summary('summary.h5')
+            else:
+                summary = Summary(Path(temp_dir) / 'summary.h5')
```
Comment (yardasol) [inline openmc/deplete/microxs.py:104]:
```suggestion
            # summary.h5 file is in the current working directory
            summary_path = 'summary.h5'
            if model.is_initialized:
                summary = Summary(summary_path)
            else:
                summary_path = (Path(temp_dir) / summary_path).as_posix()
                summary = Summary(summary_path)
```

**Review by yardasol — CHANGES_REQUESTED** [review]

File: openmc/deplete/microxs.py (hunk 1 above)
Comment (paulromano) [reply openmc/deplete/microxs.py:104]: I took a slightly different approach by just adding support for `Path` objects in `Summary`

**Comment by paulromano** [comment]
It just occurred to me that the problem is a little deeper — if you call `model.init_lib` prior to calling `MicroXS.from_model`, it won't pick up the changes that are made to the materials or tallies since those were effectively made after initialization. This is why it fails to find the tallies once it loads the statepoint (they're not there). I'm going to have to think about how best to handle this as it might require some changes in the `Model` class itself. For now, I'm going to close this PR and we'll revisit later.

**Pull request closed without merging**